flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobSubmitHandlerWithOptions.java

```diff
@@ -113,30 +110,16 @@ private void setSavepointRestoreSettings(
 
     private void setOperatorParallelisms(
             JobGraph jobGraph, Map<String, Integer> operatorParallelismChangeMap) {
-        if (jobGraph.getNumberOfVertices() != operatorParallelismChangeMap.size()) {
-            throw new CompletionException(
-                    new RestHandlerException(
-                            "Operator parallelism change map must contain all vertices of corresponding job graph.",
-                            HttpResponseStatus.BAD_REQUEST));
-        }
-
-        jobGraph.getVertices()
-                .forEach(v -> setParallelismForJobVertex(v, operatorParallelismChangeMap));
-    }
-
-    private void setParallelismForJobVertex(
-            JobVertex jobVertex, Map<String, Integer> operatorParallelismChangeMap) {
-        String jobVertexId = jobVertex.getID().toHexString();
-        if (!operatorParallelismChangeMap.containsKey(jobVertexId)) {
-            throw new CompletionException(
-                    new RestHandlerException(
-                            String.format(
-                                    "Vertex ID %s of job graph not found in provided operator parallelism change map. "
-                                            + "Map must contain all job vertices of job graph.",
-                                    jobVertexId),
-                            HttpResponseStatus.BAD_REQUEST));
+        for (Map.Entry<String, Integer> override : operatorParallelismChangeMap.entrySet()) {
+            String vertexId = override.getKey();
+            Integer parallelism = override.getValue();
+            JobVertex jobVertex = jobGraph.findVertexByID(JobVertexID.fromHexString(vertexId));
+            if (jobVertex != null) {
+                jobVertex.setParallelism(parallelism);
+            } else {
+                log.warn("JobGraph changed. Task {} not present anymore. Skipping parallelism override {}", vertexId, parallelism);
```

Comment:
Maybe let's also add a warning when `jobGraph.getNumberOfVertices() == operatorParallelismChangeMap.size()` is not true anymore? 

If I remember correctly, when additional vertices are added, the original vertex ids might still all be present in the job graph.

Reply:
Size differences are now expected because we do not specify all vertices anymore, e.g. sources vertices. So the only condition which we can detect is when there are less task vertices than overrides which is handled here.